kliff/models/kim.py

```diff
@@ -599,48 +600,32 @@ def set_one_opt_param(self, name: str, settings: List[List[Any]]):
         except RuntimeError:
             raise kimpy.KimPyError("Calling `kim_model.clear_then_refresh()` failed.")
 
-        # reset influence distance in case it changes
+        # reset influence distance in case it changes and get latest parameters
         self.init_influence_distance()
 
-    def update_model_params(self, params: Sequence[float]):
+    def update_model_params(
+        self, params: Union[np.ndarray, List[Union[float, int, Parameter]]]
+    ):
         """
         Update optimizing parameters (a sequence used by the optimizer) to the kim model.
         """
 
         # update from opt params to model params
         super().update_model_params(params)
 
-        # only update optimizing params
-        if self.params_transform is None:
-            # update from model params to kim params
-            n = self.get_num_opt_params()
-            for i in range(n):
-                _, value, p_idx, c_idx = self.get_opt_param_name_value_and_indices(i)
+        # update from model params to kim params
+        for name, param in self.model_params.items():
+            p_idx = param.index
+            # Note: only use `get_numpy_array` here, as model
```

Comment:
The comment can be removed, since we have `_model_space` and `_param_space` explicitly added to the names?